orchagent/vrrporch.h

```diff
@@ -0,0 +1,83 @@
+#ifndef VRRPORCH_H
+#define VRRPORCH_H
+
+#include <string>
+#include <sstream>
+#include "orch.h"
+#include "request_parser.h"
+#include "ipaddress.h"
+#include "ipaddresses.h"
+#include "ipprefix.h"
+#include "macaddress.h"
+
+extern sai_object_id_t gVirtualRouterId;
+
+struct vrrp_key_t
+{
+    std::string port_name;
+    IpPrefix ip_addr;
+
+    vrrp_key_t() = default;
+
+    vrrp_key_t(std::string portName, IpPrefix ipAddr)
+    {
+        port_name = portName;
+        ip_addr = ipAddr;
+    };
+
+    bool operator== (const vrrp_key_t& rhs) const
+    {
+        if (!(ip_addr == rhs.ip_addr) || port_name != rhs.port_name)
```

Comment:
Shouldn't "!(ip_addr == rhs.ip_addr)" be "(ip_addr != rhs.ip_addr)" similar to other case?

Reply:
IpPrefix base class has operator == and no operator !=. Hence the code is in this fashion.